File: Java/백준/Silver/10773. 제로/제로.java
import java.util.Scanner;
import java.util.Stack;

public class Main {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		Stack<Integer> stack = new Stack<>();
		int T = scan.nextInt();
		for(int i = 0; i < T; i ++) {
			int n = scan.nextInt();
			if(n != 0) {
				stack.push(n);
			} else stack.pop();
		}
		int sum = 0;
		for(int i = 0; i < stack.size(); i ++)
			sum += stack.get(i);
		System.out.println(sum);
	}

}

File: Kotlin (JVM)/백준/Silver/10773. 제로/제로.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>){
    //var sb: StringBuilder = StringBuilder()
    var scan: BufferedReader = BufferedReader(InputStreamReader(System.`in`))
    var stack: Stack<Int> = Stack()
    var sum: Int = 0
    var N: Int = scan.readLine().toInt()
    for (i: Int in 0 until N){
        var number: Int = scan.readLine().toInt()
        if (number != 0) {
            stack.push(number)
            sum += number
        }
        else {
            sum -= stack.pop()
        }
    }
    println(sum)
}